src/libraries/sortedLinkedList.se

```diff
@@ -0,0 +1,164 @@
+data comparor
+data collection[](exists, prev, next)
+data count
+data head
+data tail
+data owner
+data initialized
+
+inset('../macros/require.sem')
+
+extern comparisonProviderInterface: [compare:[int256, int256]:int256]
+
+def init():
+    self.owner = msg.sender
+
+def any():
+    require(msg.sender == self.owner or msg.sender == self)
+
+def initialize(comparor: address):
+    require(not self.initialized)
+    self.initialized = 1
+    self.comparor = comparor
+    return(1)
+
+def add(item: int256, hints: arr):
+    require(item != 0)
+
+    if (self.contains(item)):
+        self.remove(item)
+
+    # If this is the only node it is the head / tail and we can early out
+    if (self.count == 0):
+        self.head = item
+        self.tail = item
+        self.collection[item].exists = 1
+        self.count += 1
+        return(1)
+
+    # Use provided hints to get the starting node for traversal
+    nodeValue = self.getBestHint(hints, item)
+
+    # Do traversal to find the insertion point
+    while(self.hasPrev(nodeValue) and self.comparor.compare(nodeValue, item) >= 0):
+        nodeValue = self.getPrev(nodeValue)
```

Comment:
This seems good so I'll do these suggestions.

I'll try to make my serpent code follow this sort of pattern more in the future too.